src/Mono.Android/Xamarin.Android.Net/AndroidMessageHandler.cs

```diff
@@ -157,13 +157,19 @@ public CookieContainer CookieContainer
 
 		public bool CheckCertificateRevocationList { get; set; } = false;
 
-		X509TrustManagerWithValidationCallback.Helper? _callbackTrustManagerHelper = null;
+		ServerCertificateCustomValidator? _serverCertificateCustomValidator = null;
 
 		public Func<HttpRequestMessage, X509Certificate2?, X509Chain?, SslPolicyErrors, bool>? ServerCertificateCustomValidationCallback
 		{
-			get => _callbackTrustManagerHelper?.Callback;
+			get => _serverCertificateCustomValidator?.Callback;
 			set {
-				_callbackTrustManagerHelper = value != null ? new X509TrustManagerWithValidationCallback.Helper (value) : null;
+				if (value is null) {
+					_serverCertificateCustomValidator = null;
+				} else if (_serverCertificateCustomValidator is null) {
+					_serverCertificateCustomValidator = new ServerCertificateCustomValidator (value);
+				} else {
+					_serverCertificateCustomValidator.Callback = value;
+				}
```

Comment:
Sorry to be late to the discussion, but I don't think we should do it. Please see my [comment](https://github.com/xamarin/xamarin-android/pull/7246#discussion_r946579603).